### DISCUSSION TITLE
Support custom component

### DISCUSSION THREAD

**Discussion by Mister-Hope** [DreamOfIce/vuepress-plugin-i18n#10] [Ideas]
I think we can add support for custom component, this also means we should provide client composables to provide common logic.

**Comment by DreamOfIce** [comment]
I don't fully understand, can you describe it in more detail?

**Reply by Mister-Hope** [reply]
People can build their own `<I18nTip />` if they want, they just need to provide a component name to the plugin, e.g.: `MyTranslationTip`.

The plugin should render `<MyTranslationTip>` for users.

For composables, the plugin may need to expose a composable with `vuepress-plugin-i18n/client`, so that people can build a component easily like this:

```html
<!-- MyTranslationTip.vue -->
<script setup lang="ts">
import { useI18nData } from 'vuepress-plugin-i18n/client';

const { isOutdated, isUntranslate, locale, translateLink ... } = useI18nData();
</script>

<template>
  <!-- A highly customized component ... -->
</template>

<style>
/* ... */
</style>
```

I don't think a simple built-in tip can fit everyone.

**Comment by Mister-Hope** [comment]
BTW, honestly speaking, I am planning to build a similar (but much more powerful) plugin with multiple locales, and then I found out the plugin name is already taken by you.

Your plugin is lack of many features I need.

**Reply by DreamOfIce** [reply]
> Your plugin is lack of many features I need.

最早我做这个插件主要是为了我的一个文档项目，所以只包含了我需要的一些功能
另外，这是我第一次制作Vuepress插件，基本上是边学边做，加上中学生时间有限，确实显得十分简陋

> BTW, honestly speaking, I am planning to build a similar (but much more powerful) plugin with multiple locales, and then I found out the plugin name is already taken by you.

如果您愿意的话，我很乐意给予您权限共同维护这个插件，这点我相信您的能力

**Reply by DreamOfIce** [reply]
Of course, I will try to improve this plugin as much as I can

**Reply by DreamOfIce** [reply]
> 如果您愿意的话，我很乐意给予您权限共同维护这个插件，这点我相信您的能力

@Mister-Hope  What do you think of this?

**Reply by Mister-Hope** [reply]
I will try to rebuild this plugin tomorrow

**Reply by Mister-Hope** [reply]
You can check the refactor branch of my forked repo to see the progress, and I  may move the final code into vuepress-theme-hope/vuepress-theme-hope.

**Comment by DreamOfIce** [comment]
Completed
